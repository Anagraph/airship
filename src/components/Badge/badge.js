import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import { colors } from '../../constants';
import CloseIcon from '../Icons/close';

const StyledBadge = styled.li`
  background: ${(props) => props.color};
  border-radius: 100px;
  font: 400 12px/20px 'Roboto';
  color: ${colors.type01};
  letter-spacing: 0;
  list-style: none;
  display: inline-flex;
  align-items: center;
  padding: 0 8px;

  button {
    cursor: pointer;
    margin: 0 0 0 8px;
    border: 0;
    box-shadow: 0;
    background: transparent;
    padding: 0;
    line-height: 1;
    outline: none;
  }
`;

const noOp = (e) => {};

const Badge = ({
  color = colors.ui03,
  closeColor = colors.ui01,
  children,
  closable = false,
  onClose = noOp,
  as = 'li'
}) => {
  const Wrapper = as ? StyledBadge.withComponent(as) : StyledBadge;
  return (
    <Wrapper color={color}>
      {children}
      {closable ? (
        <button onClick={onClose}>
          <CloseIcon width={12} height={12} color={closeColor} />
        </button>
      ) : null}
    </Wrapper>
  );
};

Badge.propTypes = {
  as: PropTypes.string,
  closable: PropTypes.bool,
  onClose: PropTypes.func,
  color: PropTypes.string,
  closeColor: PropTypes.string
};

export default Badge;
