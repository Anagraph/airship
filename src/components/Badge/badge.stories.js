import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { ThemeProvider } from 'styled-components';
import Badge from './badge';
import { theme } from '../../constants';

const CUSTOM_THEME = {
  ...theme,
  ui03: '#333',
  type01: 'white',
};

storiesOf('Badge', module)
  .add('Default', () => (
    <Badge>Store</Badge>
  ))
  .add('With custom element', () => (
    <Badge as="span">I am a span</Badge>
  ))
  .add('With custom color', () => (
    <Badge color="#FABADA">Different color</Badge>
  ))
  .add('With action', () => (
    <Badge onClose={action('closing')}>Store</Badge>
  ))
  .add('With custom theme', () => (
    <ThemeProvider theme={CUSTOM_THEME}>
      <Badge>Store</Badge>
    </ThemeProvider>
  ));
