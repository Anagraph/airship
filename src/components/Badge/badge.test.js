import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Badge from './badge';

describe('render', () => {
  it('renders without crashing', () => {
    const component = renderer.create(<Badge>Foo</Badge>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('color parameter', () => {
    const component = renderer.create(
      <Badge color="rgba(128, 182, 34, 0.24)">Foo</Badge>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('click action', () => {
    const spy = jest.fn();
    const component = shallow(
      <Badge closable onClose={spy}>
        Store 2
      </Badge>
    );

    const button = component.find('button');
    button.simulate('click');

    expect(spy).toHaveBeenCalled();
  });
});
