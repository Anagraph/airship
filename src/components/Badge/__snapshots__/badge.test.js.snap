// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render color parameter 1`] = `
.c0 {
  background: rgba(128,182,34,0.24);
  border-radius: 100px;
  font: 400 12px/20px 'Roboto';
  color: #2C2C2C;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  list-style: none;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 8px;
}

.c0 button {
  cursor: pointer;
  margin: 0 0 0 8px;
  border: 0;
  box-shadow: 0;
  background: transparent;
  padding: 0;
  line-height: 1;
  outline: none;
}

<li
  className="c0"
  color="rgba(128, 182, 34, 0.24)"
>
  Foo
</li>
`;

exports[`render renders without crashing 1`] = `
.c0 {
  background: #E2E6E3;
  border-radius: 100px;
  font: 400 12px/20px 'Roboto';
  color: #2C2C2C;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  list-style: none;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 8px;
}

.c0 button {
  cursor: pointer;
  margin: 0 0 0 8px;
  border: 0;
  box-shadow: 0;
  background: transparent;
  padding: 0;
  line-height: 1;
  outline: none;
}

<li
  className="c0"
  color={undefined}
>
  Foo
</li>
`;
